src/cassio/table/base_table.py

```diff
@@ -407,32 +407,50 @@ def _get_db_setup_cql(self, schema: Dict[str, List[ColumnSpecType]]) -> str:
         )
         return create_table_cql
 
-    def _get_create_analyzer_index_cql(
-        self, index_options: List[Tuple[str, Any]]
+    @staticmethod
+    def _get_create_index_cql(
+        index_name: str, index_column: str, index_options: List[Tuple[str, Any]]
     ) -> str:
-        index_name = "idx_body"
-        index_column = "body_blob"
-        body_index_options = []
-        for option in index_options:
-            key, value = option
-            if isinstance(value, dict):
-                body_index_options.append(f"'{key}': '{json.dumps(value)}'")
-            elif isinstance(value, str):
-                body_index_options.append(f"'{key}': '{value}'")
-            elif isinstance(value, bool):
-                if value:
-                    body_index_options.append(f"'{key}': true")
+        options_clause = ""
+        if len(index_options) > 0:
+            formatted_options = []
+            for option in index_options:
+                key, value = option
+                if isinstance(value, dict):
+                    formatted_options.append(f"'{key}': '{json.dumps(value)}'")
+                elif isinstance(value, str):
+                    formatted_options.append(f"'{key}': '{value}'")
+                elif isinstance(value, bool):
+                    if value:
+                        formatted_options.append(f"'{key}': true")
+                    else:
+                        formatted_options.append(f"'{key}': false")
                 else:
-                    body_index_options.append(f"'{key}': false")
-            else:
-                raise ValueError("Unsupported body_index_option format")
+                    raise ValueError("Unsupported index_option format")
+
+            formatted_options.sort()
 
-        create_index_cql = CREATE_INDEX_ANALYZER_CQL_TEMPLATE.format(
+            options_text = ", ".join(formatted_options)
+
+            # this is double escaped because the cql will go through
+            # another format method before being executed
+            options_clause = f"WITH OPTIONS = {{{{ {options_text} }}}}"
+
+        return CREATE_INDEX_CQL_TEMPLATE.format(
+            index_name=index_name,
+            index_column=index_column,
+            options_clause=options_clause,
+        )
+
+    @staticmethod
+    def _get_create_analyzer_index_cql(index_options: List[Tuple[str, Any]]) -> str:
```

Comment:
moved most of the functionality from this method that @cbornet introduced into the `_get_create_index_cql()` method above, so I can share it with the vector index creation.